standard/unsafe-code.md

```diff
@@ -681,6 +681,7 @@ It is an error to use a captured local variable ([§11.16.6.2](expressions.md#11
 - The token “`&`” followed by a *variable_reference* ([§9.4.4](variables.md#944-precise-rules-for-determining-definite-assignment)) to a moveable variable ([§22.4](unsafe-code.md#224-fixed-and-moveable-variables)) of an unmanaged type `T`, provided the type `T*` is implicitly convertible to the pointer type given in the `fixed` statement. In this case, the initializer computes the address of the given variable, and the variable is guaranteed to remain at a fixed address for the duration of the fixed statement.
 - An expression of an *array_type* with elements of an unmanaged type `T`, provided the type `T*` is implicitly convertible to the pointer type given in the fixed statement. In this case, the initializer computes the address of the first element in the array, and the entire array is guaranteed to remain at a fixed address for the duration of the `fixed` statement. If the array expression is `null` or if the array has zero elements, the initializer computes an address equal to zero.
 - An expression of type `string`, provided the type `char*` is implicitly convertible to the pointer type given in the `fixed` statement. In this case, the initializer computes the address of the first character in the string, and the entire string is guaranteed to remain at a fixed address for the duration of the `fixed` statement. The behavior of the `fixed` statement is implementation-defined if the string expression is `null`.
+- An expression of type other than *array_type* or `string`, provided the type implements an accessible method named `GetPinnableReference`, that method has no parameters, that method returns a `ref` or `ref readonly` variable of an *unmanaged_type* `T`, and `T*` is implicitly convertible to the pointer type given in the `fixed` statement. In this case, the initializer computes the address of the returned variable, and that variable is guaranteed to remain at a fixed address for the duration of the `fixed` statement.
```

Comment:
How about `GetPinnableReference` extension methods? The wording "the type implements" doesn't seem to cover those. [SharpLab](https://sharplab.io/#v2:C4LglgNgNAJiDUAfAAgZgATIEzoMLoG8BYAKHXMwzADth0BJAblIF9TS1MBGANkxwCihUhUrc+AJwCmAM3Q06AcSnAACjWoBDAEYQpAJVlTp1AMZSAFMAAWYAM550pgJToAvAD500uaYB0TCIUQeScyLzoAK7UdpoyUpgALOgAshb4LiHCZKIUMmAAHlIw6BYKAFToAA7uTq4EbDnkjY1AA=)

I'm surprised that additional parameters with default values, or with `params`, are not allowed; but that seems to match what the compiler does.

Reply:
I'm missing something here, the original proposal lists under drawbacks:

> `GetPinnableReference` is intended to be used only in `fixed`, but nothing prevents its use in safe code, so implementor must keep that in mind.

So why isn't it declared with the `unsafe` attribute?

Reply:
I checked with the compiler team on this comment. 

It's not meant to provide any normative statement. `GetPinnableReference` may not do anything `unsafe`, so doesn't need to be marked `unsafe`. The fact that a caller can misuse the API isn't a language issue.

We could put a warning in a note if you think it's wise.

Reply:
So we *could* add `unsafe` to avoid the "drawback" but that would be an "abuse" of the modifier... hmmm... I think we just leave it as is.